src/rtcModule/webrtcAdapter.h

```diff
@@ -290,20 +290,22 @@ class LocalStreamHandle
 };
 
 
-class VideoManager
+class VideoManager : public webrtc::VideoTrackSourceInterface
 {
 public:
     virtual ~VideoManager(){}
+    static VideoManager* Create(const webrtc::VideoCaptureCapability &capabilities, const std::string &deviceName, rtc::Thread *thread);
     virtual void openDevice(const std::string &videoDevice) = 0;
     virtual void releaseDevice() = 0;
     virtual rtc::scoped_refptr<webrtc::VideoTrackSourceInterface> getVideoTrackSource() = 0;
+    static std::set<std::pair<std::string, std::string>> getVideoDevices();
 };
 
-class CaptureModuleLinux : public webrtc::Notifier<webrtc::VideoTrackSourceInterface>, public rtc::VideoSinkInterface<webrtc::VideoFrame>, public VideoManager
+class CaptureModuleLinux : public rtc::VideoSinkInterface<webrtc::VideoFrame>, public VideoManager
 {
 public:
     explicit CaptureModuleLinux(const webrtc::VideoCaptureCapability &capabilities, bool remote = false);
-    virtual ~CaptureModuleLinux();
+    virtual ~CaptureModuleLinux() override;
```

Comment:
Since dtor is declared virtual in the base class `VideoManager`, I think that `override` here makes no difference. It's not like other virtual methods that may get their parameters changed.

Reply:
I use the `override` decorator because it removes a compilation warning